### PR TITLE
Separate mappm into its own package

### LINKED ISSUE #1783: vcm is hard to pip install
This will make it easier for external collabs to use our code outside the fv3net environment. Also, for us to use vcm e.g. in google collab or another exploratory environments.

@brianhenn I remember you mentioning that you might be interested in setting this up. Probably would want to move the `mappm` fortran stuff into a separate package since the f2py builds often fail in my experience.

### PR DESCRIPTION
**Pull request by mcgibbon**
Currently `vcm` is difficult to install and can crash on import due to its compiled component, mappm. This PR separates that component into its own package, and treats it as an optional dependency so that `vcm` may be used independent of its remapping functionality.

Significant internal changes:
- `mappm` is moved into its own package in `external/mappm`, and if not installed will only raise an exception when used

Resolves #1783

### CODE REVIEW

File: .environment-scripts/install_local_packages.sh [hunk 1]
```diff
@@ -19,6 +19,7 @@ pip install -c constraints.txt \
   -e external/report \
   -e external/synth \
   -e external/vcm \
+  -e external/mappm \
```
Comment (mcgibbon) [inline .environment-scripts/install_local_packages.sh:22]: If and when I commit m1 environment files, we can put this install inside bash logic.

File: external/vcm/tox_install.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -e -x
+
+pip install -c ../../constraints.txt numpy
```
Comment (mcgibbon) [inline external/vcm/tox_install.sh:5]: tox really doesn't want to do multiple install commands, which meant I had to use this script. For some reason it also switched to doing a "nodeps" install of vcm, which necessitated the final `pip install .`. Took about an hour of trial-and-error to get this right.

**Comment by mcgibbon** [comment]
All integration tests passed.

File: external/vcm/tox.ini [hunk 3]
```diff
@@ -5,13 +5,15 @@
 
 [tox]
 envlist = py3
+skipsdist = true
```
Comment (AnnaKwa) [inline external/vcm/tox.ini:8]: What does this new flag do?

**Review by AnnaKwa — APPROVED** [review]
LGTM

File: external/vcm/tox.ini (hunk 3 above)
Comment (mcgibbon) [reply external/vcm/tox.ini:8]:
```suggestion
```

**Pull request merged**